tests/integration/test_mongodb.py

```diff
@@ -53,6 +53,35 @@ class ExampleDto(ExampleCreationDto):
     id: str
 
 
+@pytest.mark.asyncio
+async def test_dao_find_all_with_id(mongodb_fixture: MongoDbFixture):  # noqa: F811
+    """Test using the id field as part of the mapping in find_all()"""
+
+    dao = await mongodb_fixture.dao_factory.get_dao(
+        name="example",
+        dto_model=ExampleDto,
+        dto_creation_model=ExampleCreationDto,
+        id_field="id",
+    )
+
+    # insert an example resource:
+    resource_to_create = ExampleCreationDto(field_a="test1", field_b=27, field_c=True)
+    resource_inserted = await dao.insert(resource_to_create)
+
+    assert isinstance(resource_inserted, ExampleDto)
+    assert resource_inserted.dict(exclude={"id"}) == resource_to_create.dict()
+
+    # retrieve the resource with find_all
+    resources_read = [
+        x async for x in dao.find_all(mapping={"id": resource_inserted.id})
+    ]
+    assert len(resources_read) == 1
```

Comment:
```suggestion
    assert len(resources_read) == 1
    assert resources_read[0].id == resource_inserted.id
```

Reply:
Thanks. I'll also add something to check that you can use id in find_one